state-sync/diem-data-client/src/diemnet/mod.rs

```diff
@@ -171,8 +173,10 @@ impl DiemNetDataClient {
                     .message("Unable to select next peer")
                     .error(&error))
             );
+            increment_counter(&metrics::PEER_SELECTION_ERRORS, request.get_label().into());
             error
         })?;
+        let _timer = start_timer(&metrics::REQUEST_LATENCIES, request.get_label().into());
```

Comment:
Would be nice if we can keep track latency for success responses and error responses separately. But I'm not doing that yet because
- I'm not sure we need it at the moment
- I don't know an easy/simple way to do so (the current mechanism simply just stops the timer when `_timer` is dropped IIUC?)

Reply:
Yeah, I see what you mean, the timer is stopped when it's dropped. I think this should be fine for now. If we feel like we need it in the future, we can come back and figure out how to split up the timer for successes and responses. 😄 (There's ways of manually timing and then updating the counter, IIRC).